src/full.rs

```diff
@@ -0,0 +1,293 @@
+//! Module for functions which ensure full memory safety.
+//!
+//! Functions in this module are guarded from out-of-bounds memory access as
+//! well as from unaligned access, raising errors on both cases. Moreover, only
+//! a [`PodTransmutable`](trait.PodTransmutable.html)) can be used as the
+//! transmute target, thus ensuring full safety.
+//! 
+//! Unless this was previously imposed by certain means, the functions in this
+//! module may arbitrarily fail due to unaligned memory access. It is up to the
+//! user of this crate to 
+
+use crate::Error;
+use crate::guard::{Guard, PedanticGuard, PermissiveGuard, SingleValueGuard};
+#[cfg(feature = "std")]
+use crate::pod::{guarded_transmute_pod_vec};
+use crate::pod::{PodTransmutable, guarded_transmute_pod, guarded_transmute_pod_many};
+use crate::align::check_alignment;
+
+
+/// Transmute a byte slice into a single instance of a POD.
+///
+/// The byte slice must have at least enough bytes to fill a single instance of a type,
+/// extraneous data is ignored.
+///
+/// # Errors
+///
+/// An error is raised in one of the following situations:
+///
+/// - The data does not have a memory alignment compatible with `T`. You will
+///   have to make a copy anyway, or modify how the data was originally made.
+/// - The data does not have enough bytes for a single value `T`.
+///
+/// # Examples
+///
+/// ```no_run
+/// # use safe_transmute::safe_transmute_one;
+/// # include!("../tests/test_util/le_to_native.rs");
+/// # fn main() {
+/// // Little-endian
+/// # /*
+/// assert_eq!(safe_transmute_one::<u32>(&[0x00, 0x00, 0x00, 0x01])?, 0x0100_0000);
+/// # */
+/// # assert_eq!(safe_transmute_one::<u32>(&[0x00, 0x00, 0x00, 0x01].le_to_native::<u32>()).unwrap(), 0x0100_0000);
+/// # }
+/// ```
+pub fn safe_transmute_one<T>(bytes: &[u8]) -> Result<T, Error>
```

Comment:
Requirements in brackets, as opposed to a `where` clause

Occurs on:
  *	L46
  *	L80
  *	L113
  *	L165
  *	L206
  *	L249
  *	L288

Reply:
I am not opposed to placing them in the brackets when there aren't many constraints, but this can get ugly. IMO I don't prefer this:

```rust
pub fn safe_transmute_many<T: PodTransmutable, G: Guard>(bytes: &[u8]) -> Result<&[T], Error> {
```

Over this:

```rust
pub fn safe_transmute_many<T, G>(bytes: &[u8]) -> Result<&[T], Error>
where
    T: PodTransmutable,
    G: Guard,
{
```

But oh well, it's currently your take on style here, so we'll stick to it for the time being.

Reply:
The thin ~~blue~~ line is that as long as in-brackets doesn't break before return type, `where` clauses are superfluous, and that 95-char first sig doesn't even come close to the "ideal" 128.